test/integration_test.go

```diff
@@ -873,6 +874,39 @@ func TestTerragruntOutputFromRemoteState(t *testing.T) {
 
 	assert.True(t, (strings.Index(output, "app3 output") < strings.Index(output, "app1 output")) && (strings.Index(output, "app1 output") < strings.Index(output, "app2 output")))
 }
+
+func TestTerragruntOutputFromDependency(t *testing.T) {
+	// you cannot use t.Parallel() with t.Setenv
+
+	fixtureAppPath := filepath.Join(TEST_FIXTURE_OUTPUT_FROM_DEPENDENCY, "app")
+	fixtureDepPath := filepath.Join(TEST_FIXTURE_OUTPUT_FROM_DEPENDENCY, "dependency")
+
+	s3BucketName := fmt.Sprintf("terragrunt-test-bucket-%s", strings.ToLower(uniqueId()))
+	defer deleteS3Bucket(t, TERRAFORM_REMOTE_STATE_S3_REGION, s3BucketName)
+
+	rootTerragruntConfigPath := util.JoinPath(fixtureDepPath, config.DefaultTerragruntConfigPath)
+	copyTerragruntConfigAndFillPlaceholders(t, rootTerragruntConfigPath, rootTerragruntConfigPath, s3BucketName, "not-used", TERRAFORM_REMOTE_STATE_S3_REGION)
```

Comment:
I think should be used a temporary path for `copyTerragruntConfigAndFillPlaceholders` since it is changing used files, for example, if I run locally this test - git says that was changed file `test/fixture-output-from-dependency/dependency/terragrunt.hcl`

```
--- a/test/fixture-output-from-dependency/dependency/terragrunt.hcl
+++ b/test/fixture-output-from-dependency/dependency/terragrunt.hcl
@@ -9,8 +9,8 @@ remote_state {
 
   config = {
     encrypt = true
-    bucket  = "__FILL_IN_BUCKET_NAME__"
+    bucket  = "terragrunt-test-bucket-rzobct"
     key     = "${path_relative_to_include()}/terraform.tfstate"
-    region  = "__FILL_IN_REGION__"
+    region  = "us-west-2"
   }
 }
diff --git a/test/integration_test.go b/test/integration_test.go

```

Example:
https://github.com/gruntwork-io/terragrunt/blob/master/test/integration_test.go#L847